src/model/ShadingControl.cpp

```diff
@@ -463,17 +467,29 @@ namespace model {
         return true;
       }
 
-      bool result;
+      // Check if zone of subSurface not zone of existing subSurfaces
+      if (boost::optional<Space> space = subSurface.space()) {
+        if (boost::optional<ThermalZone> thermalZone = space->thermalZone()) {
+          for (auto& subSurface2 : subSurfaces()) {
+            if (boost::optional<Space> space2 = subSurface2.space()) {
+              if (boost::optional<ThermalZone> thermalZone2 = space2->thermalZone()) {
+                if (thermalZone->handle() != thermalZone2->handle()) {
+                  LOG(Warn, briefDescription() << ": cannot add two surfaces that are in different Thermal Zones ('" << thermalZone->nameString()
+                                               << "' versus existing '" << thermalZone2->nameString() << "')");
+                  return false;
+                }
+              }
+            }
+          }
+        }
+      }
```

Comment:
I thought @shorowit said that this is actually fine now to have a multiple zones referenced in the same ShadingControl? https://github.com/NREL/OpenStudio/pull/4066#issuecomment-694289272

Reply:
I'm not an expert on this object and there are so many ways to use it that it becomes very confusing. But as far as I know, it has always been fine to have subsurfaces of multiple zones referenced by a single ShadingControl, at least under some circumstances. When we did this, we used the "OnIfScheduleAllows" [shading control type](https://bigladdersoftware.com/epx/docs/9-4/input-output-reference/group-thermal-zone-description-geometry.html#field-shading-control-type). If you instead used a control type that depends on the [thermal zone](https://bigladdersoftware.com/epx/docs/9-4/input-output-reference/group-thermal-zone-description-geometry.html#field-zone-name-29-001) specified, I don't know what would happen.